### PR TITLE
Added database connection parameters

### PR DESCRIPTION
**Pull request by horkko**
Hi Keith,

As we are using remote database server for `BIGSdb`, I thought this new options could be interesting for people like us managing database(s) from a remote server.
Let me know.

Emmanel

### CODE REVIEW

File: scripts/maintenance/create_client_credentials.pl [hunk 1]
```diff
@@ -69,8 +77,8 @@ sub main {
 		say "Client secret: $client_secret";
 	}
 	if ( $opts{'i'} || $opts{'u'} ) {
-		my $db = DBI->connect( 'DBI:Pg:dbname=' . DBASE,
-			'postgres', '', { AutoCommit => 0, RaiseError => 1, PrintError => 0 } )
+		my $db = DBI->connect( 'DBI:Pg:dbname=' . DBASE . ";host=$opts{'dbhost'};port=$opts{'dbport'}",
```
Comment (kjolley) [inline scripts/maintenance/create_client_credentials.pl]: A problem here is that previously you could run this command as the postgres user on localhost without having to provide a password. Now, if you don't use the -P option, it fails with:

DBI connect('dbname=bigsdb_auth;host=localhost;port=5432','postgres',...) failed: fe_sendauth: no password supplied at ./create_cred.pl line 80.

I 'sudo su postgres' and then run the command. I think we need a test so that if no password is provided that you don't attempt to pass an empty string to the connect method.

**Comment by kjolley** [comment]
Ideally you would also support being able to put your password in the .pgpass file (https://www.postgresql.org/docs/current/static/libpq-pgpass.html), so that this is used if you provide the username with no password.

**Comment by horkko** [comment]
Hi Keith,

It seems to work for having my `.pgpass` set with my username `tuco`.
My `.pgpass`:
```
hostname:5432:*:tuco:xxxxx
```
./create_client_credentials.pl -a curl2 -i -U tuco -H hostname
Application: curl2
Version: 
Client id: xxxxxxx
Client secret: xxxxxxx

Credentials added to authentication database.
```
I think it should work for you too if your `.pgpass` is set?

Emmanuel

**Comment by kjolley** [comment]
I think my issue is with the way postgres is configured on my machine - the way the script connected before was using UNIX domain sockets as no host/port was provided. I have different security settings for this (http://bigsdb.readthedocs.io/en/latest/installation.html#configuring-postgresql). I need to modify the code slightly so that if no host/port is provided that it continues to connect using sockets.

**Comment by kjolley** [comment]
Merged - thanks Emmanuel!

**Comment by horkko** [comment]
Thanks to you for merging it 👍
